src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -917,5 +979,305 @@ namespace UiaOperationAbstractionTests
+                        Assert::AreEqual(expectedResult.numericValue, actualResult.numericValue);
+                        Assert::AreEqual(expectedResult.booleanValue, actualResult.booleanValue);
+                        Assert::AreEqual(std::wstring_view(expectedResult.stringValue.c_str()), std::wstring_view(actualResult.stringValue.c_str()));
+                    }
+
+                    std::wstring name;
+                    int numericValue = 0;
+                    bool booleanValue = false;
+                    std::wstring stringValue;
+                };
+
+                // Ensure the correct results have been returned for:
+                // -> Default-constructible `UiaTuple`
+                {
+                    const LocalTupleResult expectedResult{ L"" /* name */, 0 /* numericValue */, false /* booleanValue */, L"" /* stringValue */ };
+                    const auto actualResult = LocalTupleResult::FromResult(defaultConstructedTuple);
+                    LocalTupleResult::TestEqual(expectedResult, actualResult);
+                }
+
+                // -> `UiaTuple` constructed out of local types
+                {
+                    const LocalTupleResult expectedResult{ L"Display is 0" /* name */, 1 /* numericValue */, true /* booleanValue */, L"LocalConstructed" /* stringValue */ };
+                    const auto actualResult = LocalTupleResult::FromResult(localConstructedTuple);
+                    LocalTupleResult::TestEqual(expectedResult, actualResult);
+                }
+
+                // -> `UiaTuple` constructed out of abstraction/wrapper values.
+                {
+                    const LocalTupleResult expectedResult{ L"Display is 0" /* name */, 2 /* numericValue */, true /* booleanValue */, L"WrapperConstructed" /* stringValue */ };
+                    const auto actualResult = LocalTupleResult::FromResult(wrapperConstructedTuple);
+                    LocalTupleResult::TestEqual(expectedResult, actualResult);
+                }
+            }
+
+            // In addition, create a test where tuples are nested in other tuples.
+            {
+                auto operationScope = UiaOperationScope::StartNew();
+
+                // Give this operation a remote context.
+                UiaElement displayElement{ focusedElement };
+                operationScope.BindInput(displayElement);
+
+                // Create a default numeric tuple.
+                const std::tuple<int> baseValueTuple{ 0 };
+                UiaTuple<UiaTuple<UiaInt>> tuple;
+
+                // Return the tuple from the operation
+                operationScope.BindResult(tuple);
+                operationScope.Resolve();
+
+                // Ensure the tuple is equal to the default numeric tuple.
+                Assert::AreEqual(std::tuple<int>(0), *tuple.GetAt<0>());
+            }
+
+            // Also, make sure that `UiaTuple` instances can hold collections.
```

Comment:
Requires: https://github.com/microsoft/Microsoft-UI-UIAutomation/issues/42

Reply:
Once this is done, the `GetOrMakeSharedPtr` change has to be expanded to the two existing collections since them being nested in `UiaTuple` would cause the problem.

They do not show the same issue as `UiaTuple` because they are created without any default values/fillers at all. Whereas `UiaTuple` requires values in its internal remote array.